src/epd2in9_v2/mod.rs

```diff
@@ -0,0 +1,438 @@
+            &[
+                start_y as u8,
+                (start_y >> 8) as u8,
+                end_y as u8,
+                (end_y >> 8) as u8,
+            ],
+        )
+    }
+
+    fn set_ram_counter(&mut self, spi: &mut SPI, x: u32, y: u32) -> Result<(), SPI::Error> {
+        self.wait_until_idle();
+        // x is positioned in bytes, so the last 3 bits which show the position inside a byte in the ram
+        // aren't relevant
+        self.interface
+            .cmd_with_data(spi, Command::SetRamXAddressCounter, &[x as u8])?;
+
+        // 2 Databytes: A[7:0] & 0..A[8]
+        self.interface.cmd_with_data(
+            spi,
+            Command::SetRamYAddressCounter,
+            &[y as u8, (y >> 8) as u8],
+        )?;
+        Ok(())
+    }
+
+    /// Set your own LUT, this function is also used internally for set_lut
+    fn set_lut_helper(&mut self, spi: &mut SPI, buffer: &[u8]) -> Result<(), SPI::Error> {
+        self.wait_until_idle();
+        self.interface.cmd_with_data(spi, Command::WriteLutRegister, buffer)?;
+        self.wait_until_idle();
+        Ok(())
+    }
+}
+
+impl<SPI, CS, BUSY, DC, RST> QuickRefresh<SPI, CS, BUSY, DC, RST>
+    for Epd2in9<SPI, CS, BUSY, DC, RST>
+where
+    SPI: Write<u8>,
+    CS: OutputPin,
+    BUSY: InputPin,
+    DC: OutputPin,
+    RST: OutputPin,
+{
+    /// To be followed immediately by `update_new_frame`.
+    fn update_old_frame(&mut self, spi: &mut SPI, buffer: &[u8]) -> Result<(), SPI::Error> {
+        self.wait_until_idle();
+        self.interface.cmd_with_data(spi, Command::WriteRam, buffer)?;
+        self.interface.cmd_with_data(spi, Command::WriteRam2, buffer)
+    }
+
+    /// To be used immediately after `update_old_frame`.
+    fn update_new_frame(&mut self, spi: &mut SPI, buffer: &[u8]) -> Result<(), SPI::Error> {
+        self.wait_until_idle();
+        //TODO original waveshare library has a hardware reset for 2 ms. But it works without reset apparently.
+        
+        self.set_lut_helper(spi, &LUT_PARTIAL_2IN9)?;
+        self.interface.cmd_with_data(spi, Command::WriteOtpSelection, &[0x00, 0x00, 0x00, 0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00])?;
+        self.interface.cmd_with_data(spi, Command::BorderWaveformControl, &[0x80])?;
+        self.interface.cmd_with_data(spi, Command::DisplayUpdateControl2, &[0xC0])?;
+        self.interface.cmd(spi, Command::MasterActivation)?;
+    
+        self.wait_until_idle();
+    
+        self.use_full_frame(spi)?;
+    
+        self.interface.cmd_with_data(spi, Command::WriteRam, buffer)?;
+        Ok(())
+    }
+
+    #[allow(unused)]
+    fn update_partial_old_frame(
+        &mut self,
+        spi: &mut SPI,
+        buffer: &[u8],
+        x: u32,
+        y: u32,
+        width: u32,
+        height: u32,
+    ) -> Result<(), SPI::Error> {
+        //TODO supported by display?
+        unimplemented!()
+    }
+    
+    #[allow(unused)]
+    fn update_partial_new_frame(
```

Comment:
Please update the function documentation to something to document the unimplemented status